### PR TITLE
Bug NoMethodError on SurveyMonkey

### PR DESCRIPTION
**Pull request by monsha**
https://rollbar.com/chattermill/huginn/items/115

### CODE REVIEW

File: app/models/agents/survey_monkey_agent.rb [hunk 1]
```diff
@@ -5,8 +5,8 @@ class SurveyMonkeyAgent < Agent
 
     UNIQUENESS_LOOK_BACK = 500
     UNIQUENESS_FACTOR = 1
-    HTTP_METHOD = "get"
-    SURVEYS_URL_BASE = "https://api.surveymonkey.net/v3/surveys"
+    HTTP_METHOD = "get".freeze
+    SURVEYS_URL_BASE = "https://api.surveymonkey.net/v3/surveys".freeze
```
Comment (OpakAlex) [inline app/models/agents/survey_monkey_agent.rb:9]: nice

File: app/services/survey_monkey_parser.rb [hunk 2]
```diff
@@ -140,16 +140,17 @@ def comment
     def full_response
       questions = response.questions
       questions.each_with_object({}) do |question, hsh|
-        details = survey.find_question(question['id'])
-        heading = details['headings']&.first['heading']
+        details = survey.find_question(question['id']) || {}
+        heading = details.present? ? details['headings']&.first['heading'] : ""
```
Comment (OpakAlex) [inline app/services/survey_monkey_parser.rb:144]: `details` always is present here.

File: app/services/survey_monkey_parser.rb [hunk 3]
```diff
@@ -140,16 +140,17 @@ def comment
...
 
         hsh[question['id']] = {
           id: question['id'],
           family: details['family'],
           subtype: details['subtype'],
           question: heading,
-          answers: answers_from_question_payload(question, details)
+          answers: details.present? ? answers_from_question_payload(question, details) : question['answers']
```
Comment (OpakAlex) [inline app/services/survey_monkey_parser.rb:151]: same logic here,  `{}.present? => true`

Comment (OpakAlex) [reply app/services/survey_monkey_parser.rb:151]: oh sorry, it will be false.

**Review by OpakAlex — APPROVED** [review]

File: app/services/survey_monkey_parser.rb (hunk 2 above)
Comment (OpakAlex) [reply app/services/survey_monkey_parser.rb:144]: sorry. you are right

Comment (monsha) [reply app/services/survey_monkey_parser.rb:144]: 👍

**Review by duboff — APPROVED** [review]
👍

**Pull request merged**